### PR TITLE
week3 database assignment

### PR DESCRIPTION
**Pull request by daxiousm**

### CODE REVIEW

File: databases/week3/meal_shearing_query.sql [hunk 1]
```diff
@@ -0,0 +1,154 @@
+USE meal_share;
+
+-- Get all meals
+
+SELECT * 
+FROM Meal;
+ 
+-- Add a new meal
+INSERT INTO meal
+VALUE(null,'Doro wot', 'chicken stew', 'Tagensvej 43','2020-08-22', 5, (175*5), '2020-08-19');
```
Comment (babakbandpey) [inline databases/week3/meal_shearing_query.sql:10]: The column id can not be null.

File: databases/week3/meal_sharing.schema.sql [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  `created_date` DATE NULL,
+  PRIMARY KEY (`id`))
+ENGINE = InnoDB;
+
+
+
+-- Table `meal_share`.`Review`
+
+CREATE TABLE IF NOT EXISTS `meal_share`.`review` (
+  `id` INT NOT NULL,
```
Comment (babakbandpey) [inline databases/week3/meal_sharing.schema.sql:40]: id must be auto_increment.

File: databases/week3/meal_shearing_query.sql [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+SELECT *
+FROM review
+WHERE id = 3;
+
+-- Update a review with any id, fx 1. Update any attribute fx the title or multiple attributes
+
+UPDATE review
+SET created_date = '2020-08-19'
+WHERE id = 12;
```
Comment (babakbandpey) [inline databases/week3/meal_shearing_query.sql:78]: no such id

File: databases/week3/meal_shearing_query.sql [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+
+-- Delete a review with any id, fx 1
+
+DELETE 
+FROM review
+WHERE id = 5;
```
Comment (babakbandpey) [inline databases/week3/meal_shearing_query.sql:84]: no such id

File: databases/week3/meal_shearing_query.sql [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+-- Add a new review 
+
+INSERT INTO review (title, description, meal_id, stars, created_date)
+VALUE('very good', 'tasty chicken stew more flavourful',12,4.5,'2020-08-18');
+
+-- Get a review with any id, fx 1
+
+SELECT *
+FROM review
+WHERE id = 3;
```
Comment (babakbandpey) [inline databases/week3/meal_shearing_query.sql:72]: no such id

File: databases/week3/meal_shearing_query.sql [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+FROM review
+WHERE id = 5;
+
+-- Additional queries
+-- Get meals that has a price smaller than a specific price fx 90
+
+SELECT *
+FROM meal
+WHERE price < 250;
+
```
Comment (babakbandpey) [inline databases/week3/meal_shearing_query.sql:92]: does not return any rows

File: databases/week3/meal_shearing_query.sql [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+SELECT meal.id,
+	   meal.title,
+       meal.max_reservation,
+       SUM(reservation.number_of_guests) AS total_guests
+FROM meal
+JOIN reservation 
+	ON Reservation.meal_id = meal.id
+	GROUP BY meal.id
+	HAVING total_guests < meal.max_reservation;
+
```
Comment (babakbandpey) [inline databases/week3/meal_shearing_query.sql:104]: does not return any rows

File: databases/week3/meal_shearing_query.sql [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+	GROUP BY meal.id
+	HAVING total_guests < meal.max_reservation;
+        
+-- Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde
+
+SELECT title
+FROM meal
+WHERE meal.title 
+	LIKE '%tibs%';
+
```
Comment (babakbandpey) [inline databases/week3/meal_shearing_query.sql:111]: does not return any rows

File: databases/week3/meal_shearing_query.sql [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+
+-- Get meals that has been created between two dates
+
+SELECT *
+FROM meal
+WHERE created_date BETWEEN '2020-01-01' AND '2020-08-20';
+
```
Comment (babakbandpey) [inline databases/week3/meal_shearing_query.sql:118]: does not return any rows

File: databases/week3/meal_shearing_query.sql [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+ 
+-- Add a new meal
+INSERT INTO meal
+VALUE(null,'Doro wot', 'chicken stew', 'Tagensvej 43','2020-08-22', 5, (175*5), '2020-08-19');
+
+-- Get a meal with any id, fx 17
+
+SELECT *
+FROM Meal
+WHERE ID = 12;
```
Comment (babakbandpey) [inline databases/week3/meal_shearing_query.sql:16]: no such id

**Review by babakbandpey — CHANGES_REQUESTED** [review]
Hello,
I see that you have no data.
How did you test that it works?

Please provide some data and test all your queries.

**Review by babakbandpey — CHANGES_REQUESTED** [review]
Hello, 

Please read my comments.

File: databases/week3/meal_sharing_query.sql [hunk 11]
```diff
@@ -0,0 +1,162 @@
+USE meal_schema;
+
+-- Get all meals
+
+SELECT * 
+FROM Meal;
+ 
+-- Add a new meal
+INSERT INTO meal
+VALUE(1,'Doro wot', 'chicken stew', 'Tagensvej 43','2020-08-22', 5, (175*5), '2020-08-19'),
```
Comment (babakbandpey) [inline databases/week3/meal_sharing_query.sql:10]: The meal table will insert the primary key id automatically. The `id` is not needed to be defined.

Comment (babakbandpey) [reply databases/week3/meal_sharing_query.sql:10]: I get this error when executing your query
`Error Code: 1136. Column count doesn't match value count at row 1`
It means the number of fields do not match with the table's

**Review by babakbandpey — CHANGES_REQUESTED** [review]
Hello,

You need to run your queries and make sure that they have no errors. Test your work and submit when you are sure it is working.
Thank you.

File: databases/week3/meal_sharing_query.sql (hunk 11 above)
Comment (daxiousm) [reply databases/week3/meal_sharing_query.sql:10]: its fixed now i tested all the queries and it worked

Comment (daxiousm) [reply databases/week3/meal_sharing_query.sql:10]: i pushed the change you can look into it

File: databases/week3/meal_sharing.schema.sql [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+-- Schema meal_share
+
+CREATE SCHEMA IF NOT EXISTS `meal_share` DEFAULT CHARACTER SET utf8 ;
+USE `meal_share` ;
+
+
+-- Table `meal_share`.`Reservation`
+
+CREATE TABLE IF NOT EXISTS `meal_share`.`reservation` (
+  `id` INT NOT NULL,
```
Comment (babakbandpey) [inline databases/week3/meal_sharing.schema.sql:11]: id needs to be `unsigned` and `auto_increment`

File: databases/week3/meal_sharing.schema.sql [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+  `contact_phonenumber` VARCHAR(45) NULL,
+  `contact_name` VARCHAR(45) NULL,
+  `contact_email` VARCHAR(45) NULL,
+  PRIMARY KEY (`id`))
+ENGINE = InnoDB;
+
+
+-- Table `meal_share`.`Meal`
+CREATE TABLE IF NOT EXISTS `meal_share`.`meal` (
+  `id` INT NOT NULL,
```
Comment (babakbandpey) [inline databases/week3/meal_sharing.schema.sql:24]: id needs to be `unsigned` and `auto_increment`

File: databases/week3/meal_sharing.schema.sql [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+  `created_date` DATE NULL,
+  PRIMARY KEY (`id`))
+ENGINE = InnoDB;
+
+
+
+-- Table `meal_share`.`Reservation_has_Meal`
+
+CREATE TABLE IF NOT EXISTS `meal_share`.`Reservation_has_Meal` (
+  `reservation_id` INT NOT NULL,
```
Comment (babakbandpey) [inline databases/week3/meal_sharing.schema.sql:54]: id must be `unsigned` and `auto_increment`

**Review by babakbandpey — CHANGES_REQUESTED** [review]
all `id`s need to be `unsigned` and `auto_increment`

File: databases/week3/m2.sql [hunk 15]
```diff
@@ -0,0 +1,240 @@
...
+CREATE TABLE IF NOT EXISTS `meal_schema`.`Meal` (
+  `id` INT UNSIGNED NOT NULL AUTO_INCREMENT,
+  `title` VARCHAR(45) NULL,
+  `description` TEXT NULL,
+  `location` VARCHAR(45) NULL,
+  `when` DATETIME NULL,
+  `max_reservation` INT NULL,
+  `price` DECIMAL NULL,
+  `created_date` DATE NULL,
+  `Reservation_id` INT UNSIGNED NOT NULL,
```
Comment (babakbandpey) [inline databases/week3/m2.sql:47]: reservation_id is not belong to this table. 
remove it.

File: databases/week3/m2.sql [hunk 16]
```diff
@@ -0,0 +1,240 @@
...
+  `Review_id` INT UNSIGNED NOT NULL,
```
Comment (babakbandpey) [inline databases/week3/m2.sql:48]: review_id is not belong to this table
remove it

File: databases/week3/m2.sql [hunk 17]
```diff
@@ -0,0 +1,240 @@
...
+  PRIMARY KEY (`id`),
+  INDEX `fk_Meal_Reservation_idx` (`Reservation_id` ASC) ,
+  INDEX `fk_Meal_Review1_idx` (`Review_id` ASC),
+  CONSTRAINT `fk_Meal_Reservation`
```
Comment (babakbandpey) [inline databases/week3/m2.sql:52]: remove

File: databases/week3/m2.sql [hunk 18]
```diff
@@ -0,0 +1,240 @@
...
+    FOREIGN KEY (`Reservation_id`)
+    REFERENCES `meal_schema`.`Reservation` (`id`)
+    ON DELETE NO ACTION
+    ON UPDATE NO ACTION,
+  CONSTRAINT `fk_Meal_Review1`
```
Comment (babakbandpey) [inline databases/week3/m2.sql:57]: remove

**Review by babakbandpey — CHANGES_REQUESTED** [review]
Hello,
There is a major problem withe meals table.
It has two foreign keys which it should not have.
The reviews table has mea_id and the reservations table has the meal_id. 
The meal can exist without reviews and reservations, but reservations and reviews can not exist without the meal_id.

You have make the changes to the tables and the queries.
Thanks

**Comment by daxiousm** [comment]
> Hello,
> There is a major problem withe meals table.
> It has two foreign keys which it should not have.
> The reviews table has the mea_id and the reservations table has the meal_id.
> The meal can exist without reviews and reservations, but reservations and reviews can not exist without the meal_id.
> 
> You have to make the changes to the tables and the queries.
> Thanks

File: databases/week3/m2.sql [hunk 19]
```diff
@@ -0,0 +1,61 @@
...
+  `description` TEXT NULL,
+  `location` VARCHAR(45) NULL,
+  `when` DATETIME NULL,
+  `max_reservation` INT NULL,
+  `price` DECIMAL NULL,
+  `created_date` DATE NULL,
+  PRIMARY KEY (`id`))
+ENGINE = InnoDB;
+ 
+
```
Comment (babakbandpey) [inline databases/week3/m2.sql:50]: The relationships of these tables are not created. The foreign keys need to be added/

**Review by babakbandpey — CHANGES_REQUESTED** [review]
Hello Daxiou,

There are still issues which are not fixed.

**Pull request merged**